### PR TITLE
Gpl534 only rename collections post import

### LINKED ISSUE #38: GPL-534 Rename samples collection after successful process
**User story**
Currently samples collection are recreated every day from lighthouse files. The way it works, it could happen that during an error in process because of wrong inputs, all samples from a center might not be loaded, so we could end up with a smaller set of samples in database than what we had yesterday.

Currently we keep a copy for every day of all samples loaded. It would be a more consistent approach to not change the current samples collection unless we are confident on the consistency of data loaded. We could do this by not renaming the current samples collection unless all data load have happened without any critical errors.

**Who are the primary contacts for this story**
Eduardo Martin

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Use the current version of samples collection when new version load has failed with an error

**Dependencies**

**Additional context**
See problem from load on Jun 11th

### PR DESCRIPTION
**Pull request by JamesGlover**
Closes #38

Changes proposed in this pull request:

* Improves test coverage of main.run
* Existing sample collection only replace by new collection if import happens without critical errors

### CODE REVIEW

**Comment by JamesGlover** [comment]
Just noticed this contains #50 (GPL-531) as well. Happy to rebase to remove it if you prefer, but its pretty much just config for the new centre.

**Comment by JamesGlover** [comment]
Rebase to remove cambridge az

**Comment by JamesGlover** [comment]
And of course, the rebase has broken the tests which assumed the cambridge-az files were in place.

File: crawler/db.py [hunk 1]
```diff
@@ -80,24 +89,77 @@ def get_mongo_collection(database: Database, collection_name: str) -> Collection
...
-    cloned_collection = f"{collection.name}_{datetime.now().strftime('%y%m%d_%H%M')}"
+    new_name = f"{collection.name}_{suffix}"
 
-    logger.debug(f"Copying '{collection.name}' to '{cloned_collection}'")
+    rename_collection(database, collection, new_name)
+    return None
+
+
+def rename_collection(database: Database, collection: Collection, new_name: str) -> None:
+    """Renames a collection to the new name.
```
Comment (emrojo) [inline crawler/db.py:108]: Is database argument needed?

File: crawler/db.py [hunk 2]
```diff
@@ -80,24 +89,77 @@ def get_mongo_collection(database: Database, collection_name: str) -> Collection
...
+    cleaned up to assist with debugging.
+
+    Arguments:
+        database {Database} -- the database of the collection to replace
+        collection {Collection} -- the collection to replace
+        timestamp {str} -- A timestamp to apply to the original and temporary collection names
+    """
+    temporary_collection_name = f"tmp_{collection_name}_{timestamp}"
+    logger.debug(f"Generating temporary collection: {temporary_collection_name}")
+    temporary_collection = get_mongo_collection(database, temporary_collection_name)
```
Comment (emrojo) [inline crawler/db.py:143]: Is get_mongo_collection creating a new collection if it doesnt exist?

File: crawler/main.py [hunk 3]
```diff
@@ -62,89 +64,88 @@ def run(sftp: bool, settings_module: str = "") -> None:
...
+                logger.debug(f"Creating compund index on '{samples_collection.full_name}'")
+                # create compound index on 'Root Sample ID', 'RNA ID', 'Result', 'Lab ID' - some data
+                #   had the same plate tested at another time so ignore the data if it is exactly the
+                #   same
+                samples_collection.create_index(
+                    [
+                        (FIELD_ROOT_SAMPLE_ID, pymongo.ASCENDING),
+                        (FIELD_RNA_ID, pymongo.ASCENDING),
+                        (FIELD_RESULT, pymongo.ASCENDING),
+                        (FIELD_RNA_ID, pymongo.ASCENDING),
```
Comment (harrietc52) [inline crawler/main.py:82]: `FIELD_RNA_ID` is duplicate with line 80

File: crawler/db.py (hunk 1 above)
Comment (JamesGlover) [reply crawler/db.py:108]: Good catch. Nope, doesn't appear to be, think I was using the method signature of the original copy approach.

File: crawler/db.py (hunk 2 above)
Comment (JamesGlover) [reply crawler/db.py:143]: Sort of, Mongo creates collections lazily as soon as something gets added to them. Its actually a bit frustrating, as it makes it surprisingly tricky to check if a collection already exists or not.

File: crawler/main.py (hunk 3 above)
Comment (JamesGlover) [reply crawler/main.py:82]: Thanks, order of indicies is important, but it looks like the first definition is taking precedence here (based on observing the actual index that gets created) so I should be able to drop the second index.

**Pull request merged**